tools/data_processing_v2_beta/processors/daq_processing.py

```
@@ -110,34 +116,42 @@ def _validate_and_extract_data(self, msg: list[float | str | DAQ_RECEIVED_MESSAG
+            timestamps = unpacked_data["relative_timestamps_nanoseconds"]
+
+            if not wrote_header:
+                sensors = sorted(data.keys())
+                writer.writerow(["Timestamp (ns) +- 10ns"] + sensors)
+                wrote_header = True
+
+            sample_count = len(timestamps)
+
+            for sensor in sensors:
+                if len(data[sensor]) != sample_count:
+                    raise ValueError("Mismatched sensor data lengths")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)